docs/src/main/asciidoc/openapi-swaggerui-guide.adoc

```diff
@@ -0,0 +1,249 @@
+    }
+
+    @DELETE
+    public Set<Fruit> delete(Fruit fruit) {
+        fruits.remove(fruit);
+        return fruits;
+    }
+}
+----
+
+Notice that there is no API specification or presentation specific code
+
+== Expose OpenAPI Specifications
+
+Quarkus proposes a `smallrye-openapi` extension compliant with https://github.com/eclipse/microprofile-open-api/[Eclipse MicroProfile OpenAPI]
+specification in order to generate your API's https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md[OpenAPI v3 specification].
+
+The first and unique step is to add the `smallrye-openapi` extension to your Quarkus application:
+----
+mvn quarkus:add-extension -Dextensions="smallrye-openapi"
+----
+
+Now we are ready to run our application:
+----
+./mvnw compile quarkus:dev
+----
+
+Once your application is started, you can make a request to the default `/openapi` endpoint:
+----
+$ curl http://localhost:8080/openapi
+openapi: 3.0.1
+info:
+  title: Generated API
+  version: "1.0"
+paths:
+  /fruits:
+    get:
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+    delete:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+components:
+  schemas:
+    Fruit:
+      properties:
+        description:
+          type: string
+        name:
+          type: string
+----
+
+Hit `CTRL+C` to stop the application.
+
+== Use Swagger UI for development
+
+When building APIs, developers want to test them quickly. https://swagger.io/tools/swagger-ui/[Swagger UI] is a great tool
+permitting to visualize and interact with your APIs. It's automatically generated via your OpenAPI specification.
+
+Quarkus proposes a `swagger-ui` extension embedding a Swagger UI page configured to find your API specification at `/openapi`.
+
+*Note*: Currently the swagger-ui page is available when Quarkus is started in a dev or test mode.
+
+The first is to add the `swagger-ui` extension to your Quarkus application:
+----
+mvn quarkus:add-extension -Dextensions="swagger-ui"
+----
+
+By default, SwaggerUI is accessible at `/swagger-ui`. You can update this path with the `quarkus-swagger-ui.path` property.
+Here you have an example of configuration:
+----
+quarkus-swagger-ui.path=/my-custom-path
+----
+
+Now we are ready to run our application:
+----
+./mvnw compile quarkus:dev
+----
+
+You can check the Swagger UI path in your application's log
+----
+00:00:00,000 INFO  [io.qua.swa.run.SwaggerUiServletExtension] Swagger UI accessible at /swagger-ui
```

Comment:
Ah, now I understand why you have this log info. Considering it only appears in dev mode, maybe it makes sense to have it.